tests/lib/rules/no-shared-component-data.js

```diff
@@ -102,7 +105,18 @@ ruleTester.run('no-shared-component-data', rule, {
           data
         }
       `,
-      parserOptions: { ecmaVersion: 6, sourceType: 'module' }
+      parserOptions
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          data: () => {
+
+          }
+        }
+      `,
```

Comment:
I suspect that this test case is not correct.
The filename is `.vue` and the code is in outside of `<script>` element.